File: JAVA/2181. Merge Nodes in Between Zeros.java
class Solution {
    public ListNode mergeNodes(ListNode head) {
     ListNode root=new ListNode();
     ListNode cur=root;
     ListNode temp=head;
     int sum=0;
     while(temp!=null){
       if(temp.val==0&&sum!=0){
         cur.next=new ListNode(sum);
         cur=cur.next;
         sum=0;
       }
       sum=sum+temp.val;
       temp=temp.next;
     }
     return root.next;
    }
}

File: C++/2181. Merge Nodes in Between Zeros.cpp
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        ListNode* p1=head->next;
        ListNode* p2=p1;
        while(p2!= NULL){
            int sum=0;
            while(p2!=NULL && p2->val!=0){
                sum+=p2->val;
                p2=p2->next;
            }
            p1->val=sum;
            p2=p2->next;
            p1->next=p2;
            p1=p1->next;
        }
        return head->next;
    }
};
